compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1733,6 +1738,10 @@ class Namer { typer: Typer =>
     }
   }
 
+  class NonInferingCompleter(original: ValOrDefDef)(ictx: Context) extends Completer(original)(ictx) {
```

Comment:
Do we need a subclass here? Wouldn't that be simpler to have a flag passed in the constructor of `Completer`?

Reply:
A flag should work. Changed.